sympy/polys/matrices/domainmatrix.py

```diff
@@ -21,8 +21,72 @@
 
 
 class DomainMatrix:
+    r"""
+    Associate Matrix with :py:class:`~.Domain`
+
+    Explanation
+    ===========
+
+    DomainMatrix uses "domain elements" for its internal representation
+    which makes it more faster for many common operations
+    than current sympy Matrix class, but this advantage makes it not
+    entirely compatible with Matrix.
+    DomainMatrix could be found analogous to numpy arrays with "dtype".
+    In the DomainMatrix, each matrix has a domain such as ZZ(Integers)
+    or QQ<sqrt(2)>[Algebraic Number Field].
```

Comment:
You should use cross-referencing hyperlinks to point to other parts of the documentation e.g.
```
:py:class:`~.Domain`
```
will give a link in the docs that leads to here:
https://docs.sympy.org/dev/modules/polys/domainsref.html#sympy.polys.domains.domain.Domain

If you look in `doc/src/modules/polys/domainsintro.rst` you can see examples of other kinds of cross referencing such as
```
:ref:`ZZ`
```
to refer to the ZZ domain which links to here:
https://docs.sympy.org/dev/modules/polys/domainsref.html#zz

Reply:
Is there some sort of guide for making `rst` files in sympy. So that I can follow the correct format

Reply:
Here:
https://docs.sympy.org/latest/documentation-style-guide.html

That also links to other things like the docs for rst and sphinx.

Reply:
Thanks. I'll start adding things to the `rst` file